File: moviejava/app/src/main/java/com/example/moviejava/ui/person/PersonViewModel.java
package com.example.moviejava.ui.person;

import androidx.lifecycle.LiveData;
import androidx.lifecycle.MutableLiveData;

import com.example.moviejava.data.remote.people.PeopleRepository;
import com.example.moviejava.model.Person;
import com.example.moviejava.model.Results;
import com.example.moviejava.model.State;
import com.example.moviejava.ui.base.BaseViewModel;

import java.util.ArrayList;
import java.util.List;

import javax.inject.Inject;

import dagger.hilt.android.lifecycle.HiltViewModel;

@HiltViewModel
public class PersonViewModel extends BaseViewModel {
    private final PeopleRepository repository;

    @Inject
    public PersonViewModel(PeopleRepository repository) {
        this.repository = repository;
    }

    private final MutableLiveData<State<Person>> _person = new MutableLiveData<>();
    public LiveData<State<Person>> person = _person;

    private final List<Results> list = new ArrayList<>();

    private final MutableLiveData<State<List<Results>>> _movie = new MutableLiveData<>();
    public LiveData<State<List<Results>>> movie = _movie;

    public void getData(int id) {
        launchRxSingle(repository.getPerson(id),result->_person.postValue(State.success(result)),null);

        launchRxSingle(repository.getMovieByPerson(id),result->{
            list.addAll(result.results);
            _movie.postValue(State.success(list));
        },null);
    }
}

File: movie/app/src/main/java/com/example/movie/ui/person/PersonViewModel.kt
package com.example.movie.ui.person

import android.util.Log
import androidx.lifecycle.LiveData
import androidx.lifecycle.MutableLiveData
import com.example.movie.data.remote.people.PeopleRepository
import com.example.movie.model.Person
import com.example.movie.model.Results
import com.example.movie.model.State
import com.example.movie.ui.base.BaseViewModel
import dagger.hilt.android.lifecycle.HiltViewModel
import javax.inject.Inject

@HiltViewModel
class PersonViewModel @Inject constructor(private val repository: PeopleRepository) :
    BaseViewModel() {
    private var _person = MutableLiveData<State<Person>>()
    val person: LiveData<State<Person>>
        get() = _person
    private var list = mutableListOf<Results>()
    private var _movie = MutableLiveData<State<List<Results>>>()
    val movie: LiveData<State<List<Results>>>
        get() = _movie

    fun getPerson(id: Int) {
        launchViewModel(repository.getPerson(id = id), {
            _person.postValue(State.success())
        }, {
            Log.e("err", "$it")
        })
        launchViewModel(repository.getMovieByPerson(id = id), {
            list.addAll(it.results)
            _movie.postValue(State.success(list))
        }, {
            Log.e("err", "$it")
        })

    }
}